File: bubblesort.cpp
#include <iostream>
using namespace std;
void bubblesort(int arr[], int n)
{
    if (n == 0 || n == 1)
    {
        return;
    }
    for (int i = 0; i < n - 1; i++)
    {
        if (arr[i] > arr[i + 1])
        {
            swap(arr[i], arr[i + 1]);
        }
    }
    bubblesort(arr, n - 1);
}
int main()
{
    int arr[5] = {2, 5, 1, 6, 9};
    bubblesort(arr, 5);
    for (int i = 0; i < 5; i++)
    {
        cout << arr[i] << " ";
    }
    return 0;
}

File: Java/BubbleSort.java
// package Sorting_Algo;

/****************************   Bubble Sort 

       ->  Algorithm

    bubbleSort(array, sizeOfArray)
    for i <- 1 to indexOfLastUnsortedElement-1
    if leftElement > rightElement
      swap leftElement and rightElement
    end bubbleSort


       -> Time Complexity 
          Best	    O(n)
          Worst	    O(n2)
          Average	O(n2)

       -> Space Complexity	O(1)
*/

import java.util.Arrays;

public class BubbleSort {

    public static int[] bubbleSort(int[] arr, int n) {
        
        // loop to access each array element
        for (int i = 0; i < n - 1; i++) {

            // loop to compare adjacent elements
            for (int j = 0; j < n - i - 1; j++) {
            
                // compare two array elements
                if (arr[j] > arr[j + 1]) {
            
                    // swap arr[j+1] and arr[j]
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
        return arr; // return sorted array
    }
    public static void main(String[] args) {

        int[] arr = {34, 10, 43, 2, 6, 11, 8};
        int n = arr.length;

        System.out.println("The Array in Ascending order:");
        System.out.println(Arrays.toString(bubbleSort(arr, n)));
    }
}
